### PR TITLE
feat:Create Unit Test Enum Food Types

### LINKED ISSUE #27: Create Unit Test Enum Food Types
Description:
This issue involves developing a set of unit tests for the FoodTypes enum to ensure that each variant is correctly defined and behaves as expected. The FoodTypes enum is used to categorize various types of food within the game, which may include categories. Comprehensive testing of this enum will help prevent regressions and confirm that comparisons and any future conversions work correctly.

Key Testing Objectives:

- Instantiation and Variant Integrity:
Verify that each variant can be instantiated without errors.
Confirm that the enum values are correctly represented and accessible.

- Extensibility Considerations:
Provide test cases that can be easily extended if new variants are added to the enum later.

### PR DESCRIPTION
**Pull request by pheobeayo**
# Pull Request Overview

## 📝 Summary
Created unit tests enum food types

### Related Issues
- Closes #27 

### Type of Change
Mark with an `x` all the checkboxes that apply (like `[x]`).
- [ ] 📝 Documentation (updates to README, docs, or comments)
- [ ] 🐛 Bug fix (non-breaking change which fixes an issue)
- [x] 👌 Enhancement (non-breaking change which adds functionality)
- [ ] 💥 Breaking change (fix or feature that would cause existing functionality to change)

### What's Changed
Provide a clear and concise description of what you changed and why.

### Technical Notes
Created unit tests enum food types, this comprehensive testing of this enum will help prevent regressions and confirm that comparisons and any future conversions work

## 🔧 Tests Results
Describe the tests you performed to verify your changes.

### Test Coverage
- [x] Unit Tests
- [ ] Integration Tests
- [ ] Manual Testing

### Evidence
image

### Testing Notes
Include any special testing considerations or edge cases checked.

## 🔜 Next Steps
Indicate actions or improvements to be taken after this PR, if applicable.

### CODE REVIEW

**Review by coxmars — CHANGES_REQUESTED** [review]
Apply git pull because I refactored the entire project's structure, it means you should place this tests inside the food type file in types/

File: scripts/tests/test_food_type.cairo [hunk 1]
```diff
@@ -0,0 +1,52 @@
+#[cfg(test)]
+mod tests {
+    use super::*;
+
+    #[cfg(test)]
+    fn test_food_type_instantiation() {
```
Comment (coxmars) [inline scripts/tests/test_food_type.cairo:6]: Also, for every test this is the way to do it instead: #[cfg(test)] use:

```
#[cfg(test)]
mod tests {
    use starknet::{contract_address_const};
    
    #[test]
    #[available_gas(1000000)]
    fn test_battle_initialization() {
    }

Comment (pheobeayo) [reply scripts/tests/test_food_type.cairo:6]: Okay, I will fix the changes

**Review by coxmars — CHANGES_REQUESTED** [review]
Hi @pheobeayo now it looks good, but fix your conflicts and we can merge 🫡

**Comment by pheobeayo** [comment]
I deleted this PR because of the conflicts, which were difficult to resolve in this PR. 
I have opened another PR to solve this issue here https://github.com/AkatsukiLabs/DojoByExample/pull/70
@coxmars
